src/OnboardingSPA/pages/Steps/DesignHomepageMenu/index.js

```diff
@@ -0,0 +1,175 @@
+            selectedGlobalStyle = globalStyles.body.filter(
+                (globalStyle) =>
+                    globalStyle.title === currentData.data.theme.variation
+            )[0];
+        } else {
+            selectedGlobalStyle = globalStyles.body[0];
+        }
+        updatePreviewSettings(
+            useGlobalStylesOutput(selectedGlobalStyle, storedPreviewSettings)
+        );
+        if (selectedGlobalStyle)
+            setGlobalStyle(selectedGlobalStyle);
+
+        var makeHomepagePattern = [];
+
+        for (let key in homepages_list) {
+            var homepage_pattern_array = homepages_list[key];
+            var patternData = '';
+            homepage_pattern_array.forEach((pattern_name) => {
+                homepagePatternData?.body.forEach((homepage_pattern_data) => {
+                    if (homepage_pattern_data.slug === pattern_name)
+                        patternData += homepage_pattern_data.content
+                })
+            })
+            makeHomepagePattern.push(patternData);
+        }
+        setHomepagePattern(makeHomepagePattern);
+
+        if(currentData?.data['sitePages'].length !== 0)
+            setSelectedHomepage(homepagePatternList?.indexOf(currentData?.data['sitePages']['homepage']));
+        else{
+            currentData.data['sitePages'] = {
+                ...currentData.data['sitePages'],
+                'homepage': homepagePatternList[0]
+            };
+            setCurrentOnboardingData(currentData);
+        }
+        setisLoaded(true);
+    }
+
+    function saveDataForHomepage(idx) {
+        setSelectedHomepage(idx);
+        currentData.data['sitePages'] = {
+            ...currentData.data['sitePages'],
+            'homepage': homepagePatternList[idx]
+        };
+        setCurrentOnboardingData(currentData);
+    }
+
+    useEffect(() => {
+        if (!isLoaded)
+            getHomepagePatternsData();
+    }, [isLoaded]);
+
+    function buildHomepagePreviews() {
+
+        var homepageList = [];
+        if (homepagePattern) {
+            homepagePattern?.forEach((homepage, idx) => {
+                if(homepage)
+                    homepageList.push(
+                        <div
+                            className='homepage_preview__list'
+                        >
+                            <LivePreviewSelectableCard
+                                className={'homepage_preview__list__item'}
+                                selected={idx == selectedHomepage}
+                                blockGrammer={homepage}
+                                viewportWidth={1200}
+                                styling={'custom'}
+                                previewSettings={globalStyle}
+                                overlay={false}
+                                onClick={() => saveDataForHomepage(idx)}
+                            />
+                        </div>
+                    );
+            });
+        }
+        return homepageList;
```

Comment:
We can skip the `var homepageList` and the `if's` to use `?`'s and `return's` wherever possible similar to `buildPreviews` here? https://github.com/newfold-labs/wp-module-onboarding/blob/e32997141435f4666c6529d1c65f62e2090205ac/src/OnboardingSPA/pages/Steps/DesignThemeStyles/Menu/index.js#L82